### ISSUE TITLE
Proposal: New data source for Remote TLS cert and CA chain

### ISSUE THREAD

**Issue by micahhausler** [hashicorp/terraform-provider-tls#52]
I'd like a new data source that would fetch remote TLS certificate information from a specified address and optional hostname (for SNI if the domain is an IP). It would include all the fields from #4, but also the certificate authority chain, and the fingerprint for the cert (and fingerprint on all certs in the CA chain).

### Terraform Version

All versions 

### Affected Resource(s)
Relates to  terraform-provider-aws for resources
- [`aws_iam_openid_connect_provider`](https://www.terraform.io/docs/providers/aws/r/iam_openid_connect_provider.html)
- [`aws_eks_cluster`](https://www.terraform.io/docs/providers/aws/d/eks_cluster.html) and the [OIDC issuer](https://github.com/terraform-providers/terraform-provider-aws/pull/10006)

### Terraform Configuration Files

Example of desired configuration

```hcl
data "aws_eks_cluster" "example" {
  name = "example"
}

# New Data Source
data "tls_remote_cert" "cluster_issuer" {
  address = "${data.aws_eks_cluster.example.identity.0.oidc.0.issuer}"
  # optional, for SNI if the endpoint is an IP
  hostname = "${data.aws_eks_cluster.example.identity.0.oidc.0.issuer}" 
  # optional
  port = 443
}

resource "aws_iam_openid_connect_provider" "example_cluster_issuer" {
  url = "${data.aws_eks_cluster.example.identity.0.oidc.0.issuer}"

  client_id_list = [
    "sts.amazonaws.com",
  ]

  thumbprint_list = [
    "${data.tls_remote_cert.cluster_issuer.authority_chain.0.fingerprint}"
  ]
}
```

### Expected Behavior

I'd want all the fields from #4, plus a field `fingerprint` which would automate the process described in the [AWS IAM docs](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc_verify-thumbprint.html) to get the SHA1 Fingerprint of the certificate. 

I would also want the `authority_chain` which is a list of the CAs in the chain, with item `0` being the root of the chain.

### References
- terraform-providers/terraform-provider-tls#4

**Comment by eugene-dounar** [comment]
@fred-vogt the Python-only script output differs from the openssl-based script. My understanding is that `ssl_sock.getpeercert(True)` would return the first cert in the chain, while the last one is needed.

**Comment by fred-vogt** [comment]
@eugene-dounar - good point.  My sample in the earlier comment isn't correct.  Will update in a bit.

**Comment by kmoe** [comment]
Implemented in #62.
